ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,342 @@
+"""
+All the flexy related classes and functionality lives here
+"""
+import logging
+import os
+import yaml
+
+import io
+import configparser
+import subprocess
+from subprocess import list2cmdline, CalledProcessError
+import shlex
+import shutil
+
+from ocs_ci.framework import config, merge_dict
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import (
+    get_ocp_version, clone_repo
+)
+from ocs_ci.ocs import exceptions
+
+logger = logging.getLogger(__name__)
+
+
+def flexy_post_processing(func):
+    """
+    Perform copying of flexy-dir to nfs mount
+    and do this only if its jenkins run
+
+    Args:
+        func : run_container function
+    """
+    # Check whether its a jenkins run
+    if is_jenkins_mount():
+        def wrapper(*args, **kwargs):
+            flexy_nfs_path = os.path.join(
+                constants.JENKINS_NFS_CURRENT_CLUSTER_DIR,
+                constants.FLEXY_HOST_DIR
+            )
+            func(*args, **kwargs)
+            if not os.path.exists(flexy_nfs_path):
+                shutil.copytree(
+                    constants.FLEXY_HOST_DIR_PATH,
+                    flexy_nfs_path
+                )
+        return wrapper
+    else:
+        return func
+
+
+def is_jenkins_mount():
+    """
+    Find if this is jenkins run based on current-cluster-dir and
+    NFS mount
+
+    Returns:
+        bool: True if this is jenkins run else False
+
+    """
+    return (
+        os.path.exists(constants.JENKINS_NFS_CURRENT_CLUSTER_DIR)
+        and os.path.ismount(constants.JENKINS_NFS_CURRENT_CLUSTER_DIR)
+    )
+
+
+class FlexyBase(object):
+    """
+    A base class for all types of flexy installs
+    """
+    def __init__(self):
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        # Host dir path which will be mounted inside flexy container
+        # This will be root for all flexy housekeeping
+        self.flexy_host_dir = os.path.expanduser(
+            constants.FLEXY_HOST_DIR_PATH
+        )
+
+        # Path inside container where flexy_host_dir will be mounted
+        self.flexy_mnt_container_dir = config.ENV_DATA.get(
+            'flexy_mnt_container_dir', constants.FLEXY_MNT_CONTAINER_DIR
+        )
+        self.flexy_img_url = config.ENV_DATA.get('flexy_img_url')
+
+        # If user has provided an absolute path for env file
+        # then we don't clone private-conf repo else
+        # we will look for 'flexy_private_conf_url' , if not specified
+        # we will go ahead with default 'flexy-ocs-private' repo
+        if not config.ENV_DATA.get('flexy-env-file'):
```

Comment:
I think, that for all other configuration variables, we are using underscores instead of dash in the key name, so no sure if it wouldn't be better to rename it to `ENV_DATA['flexy_env_file']` (but I'm not sure, if it is mandatory or not).

Reply:
done